src/patterns/patterns.py

```diff
@@ -833,8 +833,8 @@ def get_busfactor_data(self, locc_metric='change-size-cos', metric='mul-changes-
                     datetime = d['datetime'][ind]
                     author = d['unique_author'][ind]
 
+            norm_factor = len(dev_knowledge_df) # total number of files in the directory, branch or project
             d = pd.DataFrame(dev_knowledge_df.groupby(['unique_author'])['dev_knowledge'].sum())
-            norm_factor = d['dev_knowledge'].sum()
             d["dev_knowledge"] = d["dev_knowledge"].apply(lambda a: a / norm_factor)
```

Comment:
A couple comments. I'm not sure what type of scaling you are using. You appear to be dividing each value in a column by the length of the column. Is that what you want? What I was expecting to see is some type of min-max scaling where you divide every value by the max of the column. This assumes no negative values and max is not 0.

Reply:
I do remember discussing this but not sure I totally follow your code. But I'll give ok.